pkg/buffer/ringbuffer.go

```diff
@@ -0,0 +1,241 @@
+package buffer
+
+import (
+	"syscall"
+
+	msycall "github.com/Softwarekang/knetty/pkg/syscall"
+)
+
+const (
+	// 64 kb
+	defaultCacheSize = 64 * K
+	maxCacheSize     = 1 * G
+)
+
+const (
+	B = 1
+	K = 1024 * B
+	M = 1024 * K
+	G = 1024 * M
+)
+
+// RingBuffer lock-free cache for a read-write goroutine.
+type RingBuffer struct {
+	p   []byte
+	r   int
+	w   int
+	cap int
+}
+
+// NewRingBuffer .
+func NewRingBuffer() *RingBuffer {
+	return NewRingBufferWithCap(defaultCacheSize)
```

Comment:
If net IO > CPU rate, the ring may full with cache size, maybe it can be set by user~😀

Reply:
if ringbuffer is full,poller will trigger read event again.

ringbuffer size can be set by user is necessary, i will fix int next pr:+1: